File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/dialog/AlertButton.kt
package ca.logaritm.dezel.modules.dialog

import ca.logaritm.dezel.core.*

/**
 * @class AlertButton
 * @since 0.1.0
 * @hidden
 */
open class AlertButton(context: JavaScriptContext) : JavaScriptClass(context) {

	//--------------------------------------------------------------------------
	// Properties
	//--------------------------------------------------------------------------

	/**
	 * The alert button's label.
	 * @property label
	 * @since 0.1.0
	 */
	open var label: Property = Property("")

	/**
	 * The alert button's renderNode.
	 * @property style
	 * @since 0.1.0
	 */
	open var style: Property = Property("normal")

	//--------------------------------------------------------------------------
	// JS Properties
	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_label
	 * @since 0.1.0
	 * @hidden
	 */
	@Suppress("unused")
	open fun jsGet_label(callback: JavaScriptGetterCallback) {
		callback.returns(this.label)
	}

	/**
	 * @method jsSet_label
	 * @since 0.1.0
	 * @hidden
	 */
	@Suppress("unused")
	open fun jsSet_label(callback: JavaScriptSetterCallback) {
		this.label = Property(callback.value)
	}

	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_style
	 * @since 0.1.0
	 * @hidden
	 */
	@Suppress("unused")
	open fun jsGet_style(callback: JavaScriptGetterCallback) {
		callback.returns(this.style)
	}

	/**
	 * @method jsSet_style
	 * @since 0.1.0
	 * @hidden
	 */
	@Suppress("unused")
	open fun jsSet_style(callback: JavaScriptSetterCallback) {
		this.style = Property(callback.value)
	}
}

File: platforms/ios/Dezel/Dezel/Modules/Dialog/AlertButton.swift
/**
 * @class AlertButton
 * @since 0.1.0
 * @hidden
 */
open class AlertButton: JavaScriptClass {

	//--------------------------------------------------------------------------
	// MARK: Properties
	//--------------------------------------------------------------------------

	/**
	 * The alert button's label.
	 * @property label
	 * @since 0.1.0
	 */
	open var label: Property = Property(string: "")

	/**
	 * The alert button's image.
	 * @property image
	 * @since 0.6.0
	 */
	open var image: Property = Property()

	/**
	 * The alert button's style.
	 * @property style
	 * @since 0.1.0
	 */
	open var style: Property = Property(string: "normal")

	//--------------------------------------------------------------------------
	// MARK: JS Properties
	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_label
	 * @since 0.1.0
	 * @hidden
	 */
	@objc open func jsGet_label(callback: JavaScriptSetterCallback) {
		callback.returns(self.label)
	}

	/**
	 * @method jsSet_label
	 * @since 0.1.0
	 * @hidden
	 */
	@objc open func jsSet_label(callback: JavaScriptSetterCallback) {
		self.label = Property(value: callback.value)
	}

	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_image
	 * @since 0.6.0
	 * @hidden
	 */
	@objc open func jsGet_image(callback: JavaScriptGetterCallback) {
		callback.returns(self.image)
	}

	/**
	 * @method jsSet_style
	 * @since 0.6.0
	 * @hidden
	 */
	@objc open func jsSet_image(callback: JavaScriptSetterCallback) {
		self.image = Property(value: callback.value)
	}

	//--------------------------------------------------------------------------

	/**
	 * @method jsGet_style
	 * @since 0.1.0
	 * @hidden
	 */
	@objc open func jsGet_style(callback: JavaScriptSetterCallback) {
		callback.returns(self.style)
	}

	/**
	 * @method jsSet_style
	 * @since 0.1.0
	 * @hidden
	 */
	@objc open func jsSet_style(callback: JavaScriptSetterCallback) {
		self.style = Property(value: callback.value)
	}
}
